### PR TITLE
Various docXML fixes:

### PR DESCRIPTION
**Pull request by JonathanFingold**
- Make sure all non-test-projects build doc XML for the Documentation config.
- Fix all critical docXML issues (ones that cause problems for the documentation repo).
- Fix many non-critical docXML issues.
- **TODO**: review project settings for the Release config, so that the documentation pipeline continues to work properly.

### CODE REVIEW

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.csproj [hunk 1]
```diff
@@ -47,6 +47,10 @@
 		<DocumentationFile>bin\$(Configuration)\netstandard2.0\Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml</DocumentationFile>
 	</PropertyGroup>
 
+	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+	  <DocumentationFile>C:\Users\v-jofing\source\repos\botbuilder-dotnet\libraries\integration\Microsoft.Bot.Builder.Integration.ApplicationInsights.Core\Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml</DocumentationFile>
+	</PropertyGroup>
```
Comment (daveta) [inline libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.csproj]: Can you remove this?

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            Adds and configures services for Application Insights to the <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" />.
+            </summary>
+            <param name="services">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection"/> which specifies the contract for a collection of service descriptors.</param>
+            <param name="botTelemetryClient">Bot Telemetry Client that logs event information.</param>
+            <param name="instrumentationKey">If Bot Telemetry Client is using Application Insights, provide the instumentation key.</param>
+            <returns>A reference to this instance after the operation has completed.</returns>
+        </member>
+        <member name="T:Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.TelemetryBotIdInitializer">
+            <summary>
+            Initializer that sets the user ID based on Bot data.
```
Comment (daveta) [inline libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml]: > user ID bas [](start = 37, length = 12)

Sets user ID and Session ID (among other properties).

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.csproj [hunk 3]
```diff
@@ -54,6 +54,7 @@
 	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
 	  <DebugType>full</DebugType>
 	  <DebugSymbols>true</DebugSymbols>
+	  <DocumentationFile>C:\Users\v-jofing\source\repos\botbuilder-dotnet\libraries\integration\Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi\Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.xml</DocumentationFile>
 	</PropertyGroup>
```
Comment (daveta) [inline libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.csproj]: Can you remove this?

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<doc>
+    <assembly>
+        <name>Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi</name>
+    </assembly>
+    <members>
+        <member name="T:Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.TelemetryBotIdInitializer">
+            <summary>
+            Initializer that sets the user ID based on Bot data.
```
Comment (daveta) [inline libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.xml]: >ets the user I [](start = 30, length = 14)

And session ID (in addition to other bot-specific properties such as activityId)

File: libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj [hunk 5]
```diff
@@ -50,6 +50,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <DebugType>full</DebugType>
     <DebugSymbols>true</DebugSymbols>
+    <DocumentationFile>C:\Users\v-jofing\source\repos\botbuilder-dotnet\libraries\Microsoft.Bot.Builder.ApplicationInsights\Microsoft.Bot.Builder.ApplicationInsights.xml</DocumentationFile>
   </PropertyGroup>
```
Comment (daveta) [inline libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj]: Has your path in here.

File: libraries/Microsoft.Bot.Builder.Dialogs/Microsoft.Bot.Builder.Dialogs.csproj [hunk 6]
```diff
@@ -52,6 +52,7 @@
 	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
 	  <DebugType>full</DebugType>
 	  <DebugSymbols>true</DebugSymbols>
+	  <DocumentationFile>C:\Users\v-jofing\source\repos\botbuilder-dotnet\libraries\Microsoft.Bot.Builder.Dialogs\Microsoft.Bot.Builder.Dialogs.xml</DocumentationFile>
 	</PropertyGroup>
```
Comment (daveta) [inline libraries/Microsoft.Bot.Builder.Dialogs/Microsoft.Bot.Builder.Dialogs.csproj]: Your path..

File: libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj [hunk 7]
```diff
@@ -49,6 +49,7 @@
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
     <CodeAnalysisRuleSet>Microsoft.Bot.Configuration.ruleset</CodeAnalysisRuleSet>
+    <DocumentationFile>C:\Users\v-jofing\source\repos\botbuilder-dotnet\libraries\Microsoft.Bot.Configuration\Microsoft.Bot.Configuration.xml</DocumentationFile>
   </PropertyGroup>
```
Comment (daveta) [inline libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj]: Your path

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.csproj (hunk 1 above)
Comment (JonathanFingold) [reply libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.csproj]: How did my path even get in there...fixing.

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.csproj (hunk 3 above)
Comment (JonathanFingold) [reply libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.csproj]: fixing.

File: libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj (hunk 5 above)
Comment (JonathanFingold) [reply libraries/Microsoft.Bot.Builder.ApplicationInsights/Microsoft.Bot.Builder.ApplicationInsights.csproj]: fixing

File: libraries/Microsoft.Bot.Builder.Dialogs/Microsoft.Bot.Builder.Dialogs.csproj (hunk 6 above)
Comment (JonathanFingold) [reply libraries/Microsoft.Bot.Builder.Dialogs/Microsoft.Bot.Builder.Dialogs.csproj]: fixing

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml (hunk 2 above)
Comment (JonathanFingold) [reply libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.xml]: Updated in Microsoft.Bot.Builder.Integration.ApplicationInsights.Core\TelemetryBotIdInitializer.cs

File: libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.xml (hunk 4 above)
Comment (JonathanFingold) [reply libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi/Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi.xml]: Updated in Microsoft.Bot.Builder.Integration.ApplicationInsights.WebApi\TelemetryBotIdInitializer.cs

File: libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj (hunk 7 above)
Comment (JonathanFingold) [reply libraries/Microsoft.Bot.Configuration/Microsoft.Bot.Configuration.csproj]: fixing.

**Pull request merged**